src/Evolution/DgSubcell/TciStatus.hpp

```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <deque>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/DgSubcell/ActiveGrid.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace evolution::dg::subcell {
+// @{
+/// Set `status` to `1` if the cell is marked as needing subcell, otherwise set
+/// `status` to `0`. `status` has grid points on the currently active mesh.
```

Comment:
You ask good questions, sir!

The TCI status is recorded in the history for LMM steppers, while for RK (and ADER in the future) there is no history for the TCI because I'm only allowing switching from subcell back to DG at full steps. The status lives on the _active grid_, but the status is determined by the history if we are doing LMM stepping. This is because we are waiting for the shock to clear the stepper history. Does that answer all the questions?

Reply:
New question: if the status is determined by the history (as you said: waiting for the shock to clear the history), why is the status set by a single element `history.front()` rather than a check over all the history data that any of them are subcell?

Reply:
Because it is the most recent element that determines what the TCI decided at this time level. It is the status of the TCI, and not the status of the history of the TCI. I consider the history portion to be an artifact of the time stepper used, not part of the TCI that determines if the immediate solution could safely be projected to DG.

Reply:
New question: if the status is an indication of the state of the solution "right now", why does it sometime read from `active_grid` and sometimes from `history`? I understand there's no history if you're using a substep/ader method, but why not use `active_grid` always?

Reply:
because if the history says that there is still a shock, the active grid will still be subcell, even though the TCI has deemed the current time level safe. That is, the current time level is fine, but we would do bad things because the history isn't fine. Thus, the active grid is not indicative of what the TCI has decided, but rather indicative of what the TCI+time stepper history requires

Reply:
my confusion is, if the active grid is subcell, but the first item in the history is dg, why is the status set to 0 ?

Reply:
Oh I see. So you don't generally expect cases where the history says "we're shock-y now" and the active_grid is DG (this sounds like a bug), whereas you DO expect cases where the history says "we're smooth now" and active_grid is Subcell (because of shock-y earlier history). Correct?

Reply:
Yes, that's right. Here's a simple 2nd order LMM case:

We are running on subcell...
- t^n: TCI decides bad, record TCI status as `Subcell` grid, and status is `1` at `t^n`
- t^{n+1}: TCI decides all is good, record TCI status as `Dg` grid, and TCI status is `0` at `t^{n+1}`. However, since we are using a 2nd Adams-Bashforth method we need to wait for the previous entry in the history to also be `Dg`. Since `t^n` is `Subcell` the `active_grid` must remain `Subcell`
- t^{n+2}: TCI decides all is good, record TCI status as `Dg` grid, and TCI status is `0` at `t^{n+2}`. _Now_ we can switch back to DG since the entire time stepper history is `Dg`.

When using DG we won't even be tracking the history because we can always switch back to subcell, it's just switching back to DG that needs to be restricted. Does this make sense now?

Re @kidder: because the TCI decided the solution is valid, but we can't do time stepping with an LMM because the time stepper history contains a discontinuity. The TCI history really is what the TCI decided, but the active grid is not chosen _solely_ on the TCI decision at the current time, but also the TCI decision in the past.

Reply:
Okay, I'm typing this out in detail  because I am obviously confused about something...

-  We record the TCI status on the current active grid
-  The TCI status is determined completely from current information
-  The current active grid is determined based on the current TCI status and (for an `N`th-order multi-step method) the  `N-1` most recent TCI statuses 
- `tci_status` is called with the current `active_grid` as determined by the description of the previous bullet and the history of the previous `N-1` (and now I see part of my confusion, I don't know what tci_history holds... is it the active_grid of the past steps or the tci_status recorded using Dg for a 0 and Subcell for a 1?). And for clarification is tci_history.front() the oldest or latest time in the history?  I was assuming oldest...

I can't go further until those questions are answered and you tell me if any of my bullets are incorrect...

Reply:
I suggest this addition to the documentation (feel free to edit):
```
/// \note It is possible to encounter a `status` of `0`, indicating a smooth
/// solution, even though the active mesh is the subcell mesh. This can occur
/// when using a multistep time integration method like Adams Bashforth, where
/// the FD/FV scheme is used as long as _any_ of the timestepper history is
/// flagged for FD/FV subcell evolution. An example of this woul be just after a
/// shock leaves the cell: the solution is now smooth (`status` takes value
/// `0`), but the multistep method continues to take FD timesteps (`status` is
/// defined on the subcell mesh) until the entire history is flagged as smooth.
```

Possible suggestion is to make the opposite case